lib/components/Web3Context/index.tsx

```diff
@@ -10,10 +11,16 @@ interface Props {
 }
 /** Init the web3Modal and expose via react context  */
 export const Web3ContextProvider: FC<Props> = (props) => {
-  const web3ProviderState = useWeb3Modal();
-
+  const providerState = useProvider();
+  const [showModal, setShowModal] = useState(false);
+  const toggleModal = () => setShowModal((s: boolean) => !s);
```

Comment:
isnt the `boolean` type auto-inferred?
```suggestion
  const toggleModal = () => setShowModal((s) => !s);
```

Reply:
true, fixed